src/compiler/checker.ts

```diff
@@ -2238,8 +2238,8 @@ namespace ts {
             return initializer || decl;
         }
 
-        function resolveExternalModuleName(location: Node, moduleReferenceExpression: Expression): Symbol | undefined {
-            return resolveExternalModuleNameWorker(location, moduleReferenceExpression, Diagnostics.Cannot_find_module_0);
+        function resolveExternalModuleName(location: Node, moduleReferenceExpression: Expression, ignoreErrors?: boolean): Symbol | undefined {
+            return resolveExternalModuleNameWorker(location, moduleReferenceExpression, ignoreErrors ? undefined : Diagnostics.Cannot_find_module_0);

```

Comment:
Do you need to handle this at https://github.com/Microsoft/TypeScript/pull/28558/files#diff-c3ed224e4daa84352f7f1abcd23e8ccaL2273 because it can result in suggestion?

Reply:
Is that supposed to be a link to the diff this comment is on? Because it seems to be.

Reply:
Its line 2273 in the diff

Reply:
Ah, naw that position always passes `/*isError*/ false`, so only manufactures suggestions, anyway. Given that it doesn't check the `moduleNotFoundError` flag, I assume the intent is to always generate the suggestions.

Reply:
So the suggestion is raised only when the container name? Seems flaky then.

Reply:
Yeah, it probably is. I assume it was written with the estimation that the module name is resolved for all imports, rather than only-imports-with-used-exports. I didn't add the suggestion, so I'm not sure myself, though.